### DISCUSSION TITLE
Continuous logging not working on Crazyflie with micro SD card deck

### DISCUSSION THREAD

**Discussion by amitx64** [bitcraze/discussions#1234] [Q&A]
**Description:**
When using the micro SD card deck with the Crazyflie, I encountered an issue with continuous logging. Initially, upon connecting the Crazyflie with the cfclient, the micro SD card was detected, and successfully mounted. However, despite the console indicating that the configuration file was read successfully and logging was enabled (`usd: Logging to: log00`), no data was being logged to the file.

Upon deleting the log file and reconnecting the Crazyflie with the cfclient, I noticed that logging only occurred after updating the `usd.logging` parameter to `zero`, effectively stopping and restarting logging. `This raises the question of why continuous logging does not occur without stopping logging first.`
![Screenshot 2024-03-17 220756](https://github.com/bitcraze/discussions/assets/79006087/75e0e9d8-ff49-44d4-921f-3ce902becaf8)
![Screenshot 2024-03-17 220915](https://github.com/bitcraze/discussions/assets/79006087/b7bab2af-5395-467f-b8be-9936387eef09)
![Screenshot 2024-03-17 220929](https://github.com/bitcraze/discussions/assets/79006087/c7ebb240-6c68-4e3f-937e-1964e1b331f9)

**Expected behavior:**
`Continuous logging should occur without the need to stop logging first and restart it.`

**Actual behavior:**
`Continuous logging does not occur unless logging is stopped and restarted.`

**Comment by tobbeanton** [comment]
I think the problem is that the "file closing" is not written to the sdcard if usd.logging is not set to 0 to stop it. I think this also should happen when powering down though, so you might have found a bug.

**Reply by amitx64** [reply]
@tobbeanton Actually, my aim is to log data during flight. This way, I can analyze the data after the flight if the drone behaves unexpectedly.

**Reply by tobbeanton** [reply]
I've created an [issue](https://github.com/bitcraze/crazyflie-firmware/issues/1352) so we can investigate the problem. Workaround is to manually stop the logging using the usd.logging variable.

**Comment by CalvinKoung** [comment]
Hi, did you solve it ? I encounter same problem. No data in the log file ~

**Reply by ToveRumar** [reply]
Hi!

I answered in this [thread](https://github.com/orgs/bitcraze/discussions/1290) as well. 
Please come back if the workaround mentioned here does not work for you. 

The [issue](https://github.com/bitcraze/crazyflie-firmware/issues/1352) is not resolved yet.

**Reply by amitx64** [reply]
@CalvinKoung Not solved yet!

**Reply by CalvinKoung** [reply]
Or  how can I store the plotter data ?

**Reply by tobbeanton** [reply]
Workaround is to manually stop the logging using the usd.logging variable.
